react/features/analytics/AnalyticsEvents.js

```diff
@@ -1,663 +1,474 @@
+    };
+};
+
+/**
+ * Creates an event which indicates that one of the buttons in the "remote
+ * video menu" was clicked.
+ *
+ * @param {string} buttonName - The name of the button.
+ * @param {Object} attributes - Attributes to attach to the event.
+ * @returns {Object} The event in a format suitable for sending via
+ *      sendAnalytics.
+ */
+export const createRemoteVideoMenuButtonEvent
+    = function(buttonName, attributes) {
+        return {
+            action: 'clicked',
+            actionSubject: buttonName,
+            attributes,
+            source: 'remote.video.menu',
+            type: TYPE_UI
+        };
+    };
+
+/**
+ * Creates an event indicating that an action related to screen sharing
+ * occurred (e.g. it was started or stopped).
+ *
+ * @param {Object} action - The action which occurred.
+ * @returns {Object} The event in a format suitable for sending via
+ *      sendAnalytics.
+ */
+export const createScreenSharingEvent = function(action) {
+    return {
+        action,
+        actionSubject: 'screen.sharing'
+    };
+};
+
+/**
+ * The local participant failed to send a "selected endpoint" message to the
+ * bridge.
+ *
+ * @param {Error} error - The error which caused the failure.
+ * @returns {Object} The event in a format suitable for sending via
+ *      sendAnalytics.
+ */
+export const createSelectParticipantFailedEvent = function(error) {
+    const event = {
+        name: 'select.participant.failed'
+    };
+
+    if (error) {
+        event.error = error.toString();
+    }
+
+    return event;
+};
+
+/**
+ * Creates an event associated with the "shared video" feature.
+ *
+ * @param {string} action - The action that the event represents.
+ * @param {Object} attributes - Attributes to attach to the event.
+ * @returns {Object} The event in a format suitable for sending via
+ *      sendAnalytics.
+ */
+export const createSharedVideoEvent = function(action, attributes = {}) {
+    return {
+        action,
+        attributes,
+        actionSubject: 'shared.video'
+    };
+};
+
+/**
+ * Creates an event associated with a shortcut being pressed, released or
+ * triggered. By convention, where appropriate an attribute named 'enable'
+ * should be used to indicate the action which resulted by the shortcut being
+ * pressed (e.g. whether screen sharing was enabled or disabled).
+ *
+ * @param {string} action - The action that the event represents (one
+ * of ACTION_SHORTCUT_PRESSED, ACTION_SHORTCUT_RELEASED
+ * or ACTION_SHORTCUT_TRIGGERED).
+ * @param {string} shortcut - The identifier of the shortcut which produced
+ * an action.
+ * @param {Object} attributes - Attributes to attach to the event.
+ * @returns {Object} The event in a format suitable for sending via
+ *      sendAnalytics.
+ */
+export const createShortcutEvent
+    = function(action, shortcut, attributes = {}) {
+        return {
+            action,
+            actionSubject: 'keyboard.shortcut',
+            actionSubjectId: shortcut,
```

Comment:
This may be the only use of actionSubjectId in this file and in lib-jitsi-meet. Is it intentional to use it? Is it valid? I should know but I forget and I'm a lazy idiot that won't look it up.

Reply:
The doc just says it's a 'unique identifier of the action subject' and it's a string :woman_shrugging: